Make `install_base` directory accessible to Gradle build

richiware opened the pull request:
Sometimes a Gradle project does not only need to install the `*.jar` files. Maybe needs to install scripts or other elements. This PR adds a environment variable GRADLE_INSTALL_PREFIX before calling `gradle build`. The Gradle script may use it to install elements in the `colcon` installation directory.

richiware committed:
Add environment variable GRADLE_INSTALL_PREFIX

Signed-off-by: Ricardo González <ricardo@richiware.dev>

jacobperron commented on colcon_gradle/task/gradle/build.py:
```diff
@@ -141,6 +141,9 @@ async def _build(self, args, env):
         cmd += ['--stacktrace']
 
+        # Add install_base to environment in GRADLE_INSTALL_PREFIX
+        env['GRADLE_INSTALL_PREFIX'] = args.install_base
```
IIUC, this environment variable is not something Gradle typically expects. IMO, introducing this variable creates undesired coupling between Gradle projects and colcon.

Maybe there's a better way for colcon to determine what artifacts from the Gradle build to install? What we're currently doing (globbing jar files) seems brittle and limiting.

richiware replied:
Thanks for the reply.

I didn't find any Gradle's mechanism to do that. My solution is based on colcon behaviour when compiling a CMake project. colcon passes the installation directory to CMake using `CMAKE_INSTALL_PREFIX` and it is the CMake project which uses this directory to install things.

But I'm open to listen an alternative.

jacobperron committed:
Switch from Travis and AppVeyor to GitHub Actions (#32)

* Switch from Travis and AppVeyor to GitHub Actions

Leveraging existing colcon CI infrastructure.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

* Use extras_require[test] instead of tests_require

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

esteve committed:
Replace CI badges (#34)

jacobperron committed:
Prepare for first release (#33)

* Remove setup.py

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

* Add pyproject.toml

Used for building distribution files.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

* Update setup.cfg

Update maintainer info.
Specify markup type.

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

* Update README

Signed-off-by: Jacob Perron <jacob@openrobotics.org>

jacobperron committed:
0.1.0

ivanpauno committed:
Return a str in which_executable() (#35)

* Return a str in which_executable()

Signed-off-by: Ivan Santiago Paunovic <ivanpauno@ekumenlabs.com>

* Address peer review comments

Signed-off-by: Ivan Santiago Paunovic <ivanpauno@ekumenlabs.com>

Signed-off-by: Ivan Santiago Paunovic <ivanpauno@ekumenlabs.com>

cottsay committed:
0.1.1

cottsay committed:
Switch to scspell for spell checking (#37)

cottsay committed:
Update target platforms for debs, python requirement

* Bionic is in ESM and does not package importlib-metadata
* Buster does not package importlib-metadata
* Stretch has Python 3.5 and hasn't been support for a long time now
* Bookworm is the current "stable" Debian
* Upstream packages require Python 3.6, so this one should too
